tests/src/FunctionalJavascript/ParagraphsFeaturesDeleteConfirmationTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Drupal\Tests\paragraphs_features\FunctionalJavascript;
+
+/**
+ * Test display delete confirmation.
+ *
+ * @group paragraphs_features
+ */
+class ParagraphsFeaturesDeleteConfirmationTest extends ParagraphsFeaturesJavascriptTestBase {
+
+  /**
+   * Test display of delete confirmation.
+   */
+  public function testDeleteConfirmation() {
+    // Create content type with paragraphs field.
+    $content_type = 'test_delete_confirm';
+
+    // Create nested paragraph with addition of one text test paragraph.
+    $this->createTestConfiguration($content_type, 1);
+    $this->setupParagraphSettings($content_type);
+    // 1) Enable setting to show confirmation on remove action.
+    $this->enableDeleteConfirmation($content_type);
+
+    $this->drupalGet("node/add/$content_type");
+    $session = $this->getSession();
+    $page = $session->getPage();
+
+    // 1a) Check that custom remove button is present and the original remove
+    // action is hidden.
+    $is_element_hidden = $session->evaluateScript("jQuery('input[data-drupal-selector=\"field-paragraphs-0-remove\"]').parent('.visually-hidden').length === 1");
+    $this->assertEquals(TRUE, $is_element_hidden, 'Original remove button should be hidden.');
+    $this->assertSession()->elementExists('xpath', '//button[contains(@class, "paragraphs-features__delete-confirm")]');
+
+    // 1b) Trigger delete confirmation message.
+    $page->find('xpath', '//button[contains(@class, "paragraphs-dropdown-toggle")]')->click();
+    $page->find('xpath', '//button[contains(@class, "paragraphs-features__delete-confirm")]')->click();
+    // Message and buttons are shown, paragraphs inner form elements should be
+    // hidden.
+    $this->assertSession()->elementExists('css', 'div.paragraphs-features__delete-confirmation');
+    $this->assertSession()->elementExists('css', 'button.paragraphs-features__delete-confirmation__remove-button');
+    $this->assertSession()->elementExists('css', 'button.paragraphs-features__delete-confirmation__cancel-button');
+    $are_elements_hidden = $session->evaluateScript("jQuery('div[id=\"field-paragraphs-0-item-wrapper\"]').children().length - jQuery('div[id=\"field-paragraphs-0-item-wrapper\"]').children('.visually-hidden').length === 1");
```

Comment:
I think it would be better to use Drupal selector (for example `data-drupal-selector="edit-field-paragraphs-demo-0"` here and then check that parent has class `.visually-hidden`.